Console/Command/CakePHPUtilShell.php

```diff
@@ -0,0 +1,330 @@
+ * Would set the $letter, $symbol and $userId vars with the values of $this->args[0], [1] and [2].
+ * If the args were not passed by the user, they would default to: $letter = 'a', $symbol = '-' and $userId = 14;
+ *
+ * Finally, in your shell, run this small foreach loop to set the vars:
+ * foreach ($args as $arg => $val) {
+ *     ${$arg} = $val;
+ * }
+ *
+ * Or alternatively if there are only a few vars:
+ * list($someVar, $somethingElse, $userId) = array_values($args);
+ *
+ * @return array
+ */
+	public function processArgs(array $args, array $defaults = null) {
+		$vars = array();
+		foreach ($args as $id => $arg) {
+			if (array_key_exists($id, $this->args)) {
+				$vars[$arg] = $this->args[$id];
+			} else {
+				if (is_array($defaults) && array_key_exists($id, $defaults)) {
+					$vars[$arg] = $defaults[$id];
+				} else {
+					$vars[$arg] = null;
+				}
+			}
+		}
+
+		return $vars;
+	}
+
+	/**
+	 * Acts as a callback function which will be called if the SubCommand does not exist
+	 */
+	public function main() {
+		if (isset($this->command)) {
+			if (!$this->hasMethod($this->command)) {
+				$this->_exceptions[] = sprintf('The SubCommand \'%s\' you are tyring to execute does not exist!', $this->command);
+			} else {
+				$this->_exceptions[] = sprintf('For some reason this SubCommand \'%s]\' could not be ran', $this->command);
+			}
+		} else {
+			$this->_exceptions[] = 'you have to provide a SubCommand to execute!';
+		}
+
+		#TODO uncomment the line below, which echoes the help. This is used to display the getOptionsParser. Every Shell should be reformatted to the new structure first!
+		#For an example: have a look at the FeedShell
+		$this->out('-- The Help will be enabled when all Shells are restructured to the new layout -- AppShell line ' . (__LINE__ + 1));
+//        $this->out($this->getOptionParser()->help());
+		# ------------------------------------------------------------------------------------------------------------ #
+
+		# since the main() will only be called if the command does not exist, we can stop it.
+		$this->_stop();
+	}
+
+	/**
+	 * Terminates execution of the script immediately and have a look if there are exceptions whatsoever and send them to the $_mailTo
+	 *
+	 * @param int  $code     This value is used as exit code for the script (0 meaning an 'expected/healthy' termination).
+	 *
+	 * @return void
+	 */
+	protected function _stop($code = 0) {
+		# stop the timer
+		$this->_out('TIME TO COMPLETE TASK: ' . date('H\h i\m s\s', microtime(true) - $this->_startTime - 3600));
+		$this->_printWarnings();
+		// Set arg names
+		$trace = $this->_getTrace();
+		$component = ($this->command) ? $this->command : 'Unknown';
+		$args = $this->args;
+		if (!isset($this->command) || !$this->hasMethod($this->command)) {
+			# if the command is not found, or it does not exist, shift the arguments by one, because it will remain on position 0 until it has ran.
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Split this 138 characters long line (which is greater than 120 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://dev.expandonline.nl:9000/sonarqube/coding_rules#rule_key=php%3AS103)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use "//" instead of "#" to start this comment [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://dev.expandonline.nl:9000/sonarqube/coding_rules#rule_key=php%3AS2046)